pkg/monitortests/network/onpremhaproxy/monitortest.go

```diff
@@ -0,0 +1,291 @@
+		previousChangeTime := time.Time{}
+		createdIntervals := monitorapi.Intervals(allChangesForTheNode).Filter(func(eventInterval monitorapi.Interval) bool {
+			return eventInterval.Message.Reason == monitorapi.OnPremHaproxyStatusChange
+		})
+		if len(createdIntervals) > 0 {
+			previousChangeTime = createdIntervals[0].From
+		}
+		lastAvailableStatus := ""
+
+		for _, availableConditionChange := range allChangesForTheNode {
+			currentStatus := availableConditionChange.Message.Annotations[monitorapi.AnnotationStatus]
+			if currentStatus == lastAvailableStatus {
+				continue
+			}
+
+			if currentStatus == "UP" && lastAvailableStatus == "DOWN" {
+				constructedIntervals = append(constructedIntervals,
+					monitorapi.NewInterval(monitorapi.SourceHaproxyMonitor, monitorapi.Info).
+						//Locator(availableConditionChange.Locator).
+						Locator(monitorapi.Locator{Keys: map[monitorapi.LocatorKey]string{
+							monitorapi.LocatorOnPremKubeapiUnreachableFromHaproxyKey: key,
+						}}).
+						Message(monitorapi.NewMessage().Reason(monitorapi.OnPremHaproxyDetectsDown).
+							Constructed(monitorapi.ConstructionOwnerOnPremHaproxy).
+							HumanMessage(fmt.Sprintf("Kubeapi on %s is detected dead by %s", availableConditionChange.Message.Annotations[monitorapi.AnnotationNode], availableConditionChange.Locator.Keys["node"]))).
+						Display().
+						Build(previousChangeTime, availableConditionChange.From),
+				)
+			}
+
+			previousChangeTime = availableConditionChange.From
+			lastAvailableStatus = availableConditionChange.Message.Annotations[monitorapi.AnnotationStatus]
+		}
+
+		//deletionTime := time.Now()
+		//deletedIntervals := monitorapi.Intervals(allChangesForTheNode).Filter(func(eventInterval monitorapi.Interval) bool {
+		//	return eventInterval.Message.Reason == monitorapi.APIServiceDeletedInAPI
+		//})
+		//if len(deletedIntervals) > 0 {
+		//	deletionTime = deletedIntervals[0].To
+		//}
+		//if len(lastAvailableStatus) > 0 {
+		//	reason := monitorapi.APIServiceUnavailable
+		//	intervalLevel := monitorapi.Error
+		//	if lastAvailableStatus == "True" {
+		//		reason = monitorapi.APIServiceAvailable
+		//		intervalLevel = monitorapi.Info
+		//	} else if lastAvailableStatus != "False" {
+		//		reason = monitorapi.APIServiceUnknown
+		//		intervalLevel = monitorapi.Warning
+		//	}
+		//	constructedIntervals = append(constructedIntervals,
+		//		monitorapi.NewInterval(monitorapi.SourceAPIServiceMonitor, intervalLevel).
+		//			Locator(apiserviceLocator).
+		//			Message(monitorapi.NewMessage().Reason(reason).
+		//				Constructed(monitorapi.ConstructionOwnerAPIServiceLifecycle).
+		//				HumanMessage(previousHumanMesage)).
+		//			Display().
+		//			Build(previousChangeTime, deletionTime),
+		//	)
+		//}
+	}
+
+	return constructedIntervals, nil
+}
+
+func (*operatorLogAnalyzer) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
```

Comment:
This should create a failed test if collection failed for some reason

Reply:
We should let it live for a while before we decide to mark the test as Failed, opposed to current flakiness. At this moment I am not sure there is a common agreement on how widespread the outage should be for us to mark it as a Failure.

When only 1 haproxy instance sees only 1 apiserver as down, do you already call it an "outage"? From client perspective you still reach the API. And so on, and so on

Reply:
Ah, I agree, it should flake but my point is that failure here means collection failed.

>When only 1 haproxy instance sees only 1 apiserver as down, do you already call it an "outage"? From client perspective you still reach the API. And so on, and so on

No, we have separate outage monitoring, which uses both load-balancer and service network paths to reach API server. The intervals created by this test would be useful for us during investigation - but imo this shouldn't outright fail the test which a single apiserver instance is marked as down in the LB (this is expected to happen during revision rollouts)

Reply:
Ok, I may need your help then. Because what we have here is the following

* there are 3 namespaces of our interest
* max 1 of them contains our workload that has interesting logs

but at the same time this tests when running on AWS will not collect a single line of log. So I cannot easily say "if len(logs) == 0, fail".

Do we have some elegant programmatic way of doing "if platform.isOnPrem()" in this repo?